Dev

shawtylo opened the pull request:
Задания второго урока
src/lesson_2_home_work/exercise_1/dateMillis.java
src/lesson_2_home_work/exercise_2/sumTwo.java
src/lesson_2_home_work/exercise_3/sumThree.java

shawtylo committed:
lesson_1_beautify

shawtylo committed:
lesson_1_beautify_2

shawtylo committed:
lesson_2

shawtylo committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	src/lesson_1_home_work/lesson_1_git_exercises/exercise_1/sum.java
#	src/lesson_1_home_work/lesson_1_git_exercises/exercise_2/char_fool.java
#	src/lesson_1_home_work/lesson_1_git_exercises/exercise_3/sheeps.java

shawtylo committed:
Merge pull request #4 from FreeCourseJava/main

Merge pull request #3 from FreeCourseJava/dev

shawtylo committed:
Merge remote-tracking branch 'origin/dev' into dev

hyragano commented on src/lesson_1_home_work/lesson_1_git_exercises/exercise_1/sum.java:
```diff
@@ -1,18 +1,15 @@
+//// ########## lesson_1 git lesson_1_home_work.lesson_1_git_exercises.exercise_3.exercise_1 ##########
 // 1. Написать метод увеличения счета (счет и часть для увеличения int, всегда положительные), объяснить какие могут быть проблемы.
-package lesson_1_home_work.exercise_1;
+package lesson_1_home_work.lesson_1_git_exercises.exercise_1;
```
Пока пакеты не проходили, но если юзаешь. то там без _, разделяются слова точками. lesson.one.git.exercises

shawtylo replied:
Пакеты переименовал. в новом коммите

hyragano commented on src/lesson_1_home_work/lesson_1_git_exercises/exercise_1/sum.java:
```diff
@@ -1,18 +1,15 @@
-        sum = bill + partOfBill;
-        return sum;
+    public static int getFinalScore (int bill, int partOfBill){
+        return bill + partOfBill;
```
если будет int overflow, то будет бяда и можно увести счет в овердрафт жёсткий. Хуяк и чел был миллионером, стал должен

shawtylo replied:
Это задание из первого урока, там же про теор часть речь больше

hyragano commented on src/lesson_1_home_work/lesson_1_theory_exercises/exercise_4.java:
```diff
@@ -0,0 +1,43 @@
+// 4. Float 1000000000 умножить на само себя несколько раз, вывести результат каждого вычисления и объяснить, что происходит
+package lesson_1_home_work.lesson_1_theory_exercises;
+
+public class exercise_4 {
```
названия калссов CamelCase, но с заглавной. 
лучше пока просто какое-то одно слово юзать

shawtylo replied:
Названия классов переименовал.  в новом коммите

hyragano commented on src/lesson_2_home_work/exercise_1/dateMillis.java:
```diff
@@ -0,0 +1,24 @@
+        System.out.println("Результат перевода " + days + " дней, в миллисекунды, с помощью второго метода: " + secondMethod);
+    }
+
+    public static long getFirstDaysIntoMillis (int days){
```
круто конечно, но System.timeInMill лучше пока или как там его

hyragano commented on src/lesson_2_home_work/exercise_3/sumThree.java:
```diff
@@ -0,0 +1,30 @@
+    public static int getfinalScore (int bill, int partOfBill){
+        int sum = bill + partOfBill;
+        if ( bill > 0 && partOfBill > 0) {
+            if ( sum < 0 ) {
```
а если оба отрицательные? Тут нужно продумать проверку

hyragano replied:
если нужен намек, то говори, а так играемся с edge cases

shawtylo replied:
Добавил еще блок if для проверки int underwlow.  в новом коммите

shawtylo committed:
lesson_2 исправления по комментариям:
- переименовал пакеты
- переименовал классы
- 3 упражнение - добавил проверку для int underflow

shawtylo committed:
поправил описание заданий

hyragano commented on src/lesson2/home/work/exercise_3/SumThree.java:
```diff
@@ -0,0 +1,38 @@
+        System.out.println("Результат сложения " + bill + " и " + partOfBill + " равен: " + finalScore);
+
+    }
+    public static int getfinalScore (int bill, int partOfBill){
```
названиеМетодаДолжноБыть camelCase

hyragano reviewed (changes requested):
Так. Удалить не нужные коментарии. попытался поревьювить, но нойза очень много. Плз удали ненужные коментарии. И поправь один из моментов, которые я написал
